controllers/controllers.go

```diff
@@ -155,9 +155,35 @@ func (r *instanceReconciler) deconfigureInstance(node *core.Node) error {
 	return nil
 }
 
-// windowsNodePredicate returns a predicate which filters out all node objects that are not Windows nodes.
+// windowsNodeUpgradePredicate returns a predicate whose filter catches only Windows node objects that signal upgrade
+// to the latest WMCO version through a change in the node version annotation
+func windowsNodeUpgradePredicate() predicate.Funcs {
```

Comment:
Something doesn't sound right. 

There is no indication in the implementation of this function that the predicates are targeting a Node object. Any resource with the `windows` label and the `metadata.VersionAnnotation` annotation satisfies the filter conditions, hence the function name `windowsNodeUpgradePredicate` is not quite accurate. 

Not quite sure, but adding `o.GetObjectKind().GroupVersionKind().Kind == "Node"` may be helpful.


Now, if the given conditions are guaranteed to uniquely identify a Windows node you should state that in the func description.

Reply:
I see the idea behind this but given it the predicate is used only in context of watching node objects (and we only apply the version annotation to Windows node objects) I don't see it necessary to add a check on the object type. We follow the same principle in the [existing predicate function](https://github.com/openshift/windows-machine-config-operator/blob/master/controllers/controllers.go#L158-L160).